Userland/xargs.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (c) 2020, The SerenityOS developers.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <AK/StdLibExtras.h>
+#include <AK/StringBuilder.h>
+#include <LibCore/ArgsParser.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+
+int main(int argc, char** argv)
+{
+    if (pledge("stdio rpath proc exec", nullptr) < 0) {
+        perror("pledge");
+        return 1;
+    }
+
+    const char* command = nullptr;
+    const char* replace_str = "{}";
+    bool split_with_nulls = false;
+    const char* specified_delimiter = "\n";
+    Vector<const char*> arguments;
+    bool verbose = false;
+    const char* file_to_read = "-";
+
+    Core::ArgsParser args_parser;
+    args_parser.add_positional_argument(command, "Command to run", "command");
+    args_parser.add_positional_argument(arguments, "Initial arguments to pass to command", "initial_args", Core::ArgsParser::Required::No);
```

Comment:
Not important, but the convention is (aka I did this when porting stuff to `Core::ArgsParser`) to add options first, then positional args.

Reply:
Noted.
Thanks for letting me know.

Reply:
Is there a reason to differentiate the command from the arguments? Like, isn't this all just argv?

Reply:
- We don't want to replace anything in the command
- the command isn't optional

Reply:
> We don't want to replace anything in the command

Why not support it?

> the command isn't optional

Yes, so the argv as a whole isn't optional either. (Actually, Dr. POSIX says "If utility is omitted, the default shall be the echo utility.")

Reply:
Might I ask what the point of that is?
I mean, it's pretty trivial to support, but I don't see any point in a glorified `cat my-file`

Reply:
It replaces any whitespace with simple spaces for one thing. But I'm not saying this is useful, this is just what POSIX specifies.

Treating argv[0] same as the rest of argv, on the other hand, is a beautiful principle that we should follow whenever possible.

Reply:
So we should allow replacements in the command? I mean...why not.